pkg/machinepool/helper.go

```diff
@@ -198,33 +208,35 @@ func maxReplicaValidator(minReplicas int, multiAZMachinePool bool) interactive.V
 		if minReplicas > maxReplicas {
 			return fmt.Errorf("max-replicas must be greater or equal to min-replicas")
 		}
-		if multiAZMachinePool && maxReplicas%3 != 0 {
+		if replicaSizeValidationArgs.MultiAz && maxReplicas%3 != 0 {
 			return fmt.Errorf("Multi AZ clusters require that the replicas be a multiple of 3")
 		}
-		return nil
+		return ValidateClusterVersionWithMaxNodesLimit(
+			replicaSizeValidationArgs.ClusterVersion, maxReplicas, replicaSizeValidationArgs.IsHostedCp)
 	}
 }
 
-func minReplicaValidator(multiAZMachinePool bool, autoscaling bool, isHypershift bool) interactive.Validator {
+func minReplicaValidator(replicaSizeValidationArgs *ReplicaSizeValidationArgs) interactive.Validator {
```

Comment:
Could be part of the ReplicaSizeValidationArgs struct instead

Reply:
https://github.com/openshift/rosa/pull/2662#discussion_r1870186738